src/main/java/net/neoforged/neoforge/resource/ResourcePackLoader.java

```diff
@@ -208,7 +212,15 @@ private static Pack makePack(PackType packType, ArrayList<Pack> hiddenPacks) {
     }
 
     public static Pack.ResourcesSupplier createPackForMod(IModFileInfo mf) {
-        return new PathPackResources.PathResourcesSupplier(mf.getFile().getSecureJar().getRootPath());
+        return createPackForJarContents(mf.getFile().getContents());
+    }
+
+    public static Pack.ResourcesSupplier createPackForJarContents(JarContents contents) {
+        return switch (contents) {
```

Comment:
Is there any advantage to this over just always using `JarContentsPackResources`? Since, well, the underlying jar or whatever has to be open anyways, I would think.